src/LogzioShipper/file_handler.py

```diff
@@ -151,42 +161,42 @@ def _add_custom_fields_to_logzio_shipper(self) -> None:
         for custom_field in self._custom_fields:
             self._logzio_shipper.add_custom_field_to_list(custom_field)
 
-    def _is_file_csv(self, logs_sample: list) -> Optional[str]:
-        sample = '\n'.join(logs_sample)
+    def _get_csv_delimiter(self) -> str:
+        logs_sample = [self._file_stream.readline().decode("utf-8").rstrip(),
+                       self._file_stream.readline().decode("utf-8").rstrip()]
+
+        self._file_stream.seek(0)
 
         try:
-            dialect = csv.Sniffer().sniff(sample, FileHandler.CSV_DELIMITERS)
-
+            dialect = csv.Sniffer().sniff('\n'.join(logs_sample))
             return str(dialect.delimiter)
         except csv.Error:
-            return None
+            logger.error("Could not determine delimiter for the csv file - {}".format(self._file_name))
+            raise self.FormatError
 
     def _send_logs_to_logzio(self) -> None:
-        is_log_added_to_send = False
-
         for log in self._file_parser.parse_file():
-            if not self._is_log_date_greater_or_equal_filter_date(log):
-                logger.info("Log was not sent to Logzio because of filter date - {}".format(log))
+            if not self._is_log_date_greater_or_equal_date_filter(log):
+                logger.info("Log was not sent to Logz.io because of date filter - {}".format(log))
                 continue
 
             try:
```

Comment:
you already send relevant execption under each function so lets put add_log_to_send and send_to_logzio under the same try and catch

Reply:
you cant put them in the same try catch because the first try is inside a loop, adding logs. the other one is outside the loop, sending remaining logs.